### PR TITLE
Improve compression performance

### PR DESCRIPTION
**Pull request by ericmillin**
This PR changes a few things:

- Fixes a flaky `Handshake` test
- Adds EntryList `Un/MarshalPacked` methods
- Adds a buffer pool to improve compression performance

### CODE REVIEW

File: fluent/protocol/transport.go [hunk 1]
```diff
@@ -112,16 +116,47 @@ type EntryExt struct {
...
+			break
+		}
+
+		*el = append(*el, entry)
+	}
+
+	return bits, err
+}
+
+func (el EntryList) MarshalPacked() ([]byte, error) {
```
Comment (ericmillin) [inline fluent/protocol/transport.go:138]: replaces `eventStream` function

File: fluent/protocol/transport.go [hunk 2]
```diff
@@ -112,16 +116,47 @@ type EntryExt struct {
 
 type EntryList []EntryExt
 
+func (el *EntryList) UnmarshalPacked(bits []byte) ([]byte, error) {
```
Comment (ericmillin) [inline fluent/protocol/transport.go:119]: Adding this allowed the `"Correctly encodes the entries into a bytestream` test case to work

**Review by NeilTowne — APPROVED** [review]
Looks good, nice channel usage

File: fluent/protocol/transport.go [hunk 3]
```diff
@@ -40,11 +42,15 @@ func init() {
 	})
 
 	compressorPool.New = func() interface{} {
-		return &GzipCompressor{}
+		return new(GzipCompressor)
```
Comment (BernhardSchmid) [inline fluent/protocol/transport.go:45]: Just out of curiosity: what is the reason to switch to new() here instead of &{}?

**Review by BernhardSchmid — APPROVED** [review]
LGTM

**Review by gufranmirza — APPROVED** [review]

**Pull request merged**